MagickCore/xwindow.c

```diff
@@ -9708,6 +9708,8 @@ MagickPrivate void XWarning(const ExceptionType magick_unused(warning),
   (void) CopyMagickString(text,reason,MagickPathExtent);
   (void) ConcatenateMagickString(text,":",MagickPathExtent);
   windows=XSetWindows((XWindows *) ~0);
+  if (windows == NULL || description == NULL)
```

Comment:
Please remove the `description  == NULL` check because this is already done inside the method. And please change your patch to:

```c
if (windows != ((XWindows *) NULL)
   XNoticeWidget(windows->display,windows,text,(char *) description);
```